projects/onto-search/src/lib/onto-search-facet/range-facet/range-slider/range-slider.component.ts

```diff
@@ -136,21 +164,33 @@ export class RangeSliderComponent implements OnInit, AfterViewInit {
   }
 
   public onSelectionChange(): void {
-    this.selectionChange.emit(this.data.selected);
+    this.selectionChange.emit(this.data.selectedRange);
   }
 
   private updateSelection(): void {
-    this.data.selected = [];
-    this.data.facetGroupData.forEach((facet) => {
-      if (this.isInRange(facet)) {
-        this.data.selected.push(facet);
-      }
-    });
+    this.data.selectedRange = {
+      start: this.selectedMin,
+      end: this.selectedMax
+    } as SelectedRange;
     this.onSelectionChange();
   }
 
   private isInRange(facet: SearchFacetModel): boolean {
-    const value = parseInt(facet.label);
-    return value >= parseInt(this.selectedMin) && value <= parseInt(this.selectedMax);
+    const value = this.parseInt(facet.label);
+    return value >= this.parseInt(this.selectedMin) && value <= this.parseInt(this.selectedMax);
+  }
+
+  private parseInt(value: string): number {
+    try {
+      return parseInt(value);
+    } catch (e) {
+      throw new Error(`Parsing ${value} to number failed! ` + e.message);
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.onResize = function(): void {};
```

Comment:
Why do you set this? In case something still has reference to it?

Reply:
Yes - to prevent mem leaks

Reply:
Shouldn't it be set to null or undef?